wemake_python_styleguide/visitors/ast/compares.py

```diff
@@ -455,5 +455,11 @@ def _check_single_item_container(self, node: ast.AST) -> None:
             self.add_violation(InCompareWithSingleItemContainerViolation(node))
 
     def _check_wrong_comparators(self, node: ast.AST) -> None:
+        node = self._get_assigned_expr(node)
         if isinstance(node, self._wrong_in_comparators):
             self.add_violation(WrongInCompareTypeViolation(node))
+
+    def _get_assigned_expr(self, node: ast.AST) -> ast.AST:
```

Comment:
Let's better move this into `logic/walrus.py`

And one more thing: we use `compat/nodes.py` where we define nodes that appear in new `ast` releases.